[inductor] bucketing stats logging

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #173386
* #173952
* #170578
* #170575
* __->__ #170569

More logging for bucketing to compare runs 

Example of logged tlparse:
```
bucket key stats ('all_gather', ('0',)): 4 in 1 buckets of total:4
bucket key stats ('reduce_scatter', ('0', 'sum', torch.bfloat16)): 4 in 1 buckets of total:4

bucket[0] key:('all_gather', ('0',)) len:4:CollBucket(collectives=[all_gather_into_tensor, all_gather_into_tensor_1, all_gather_into_tensor_2, all_gather_into_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=12582912)
coll:all_gather_into_tensor hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_1 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_2 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_3 hiding_nodes:OrderedSet([])
bucket[1] key:('reduce_scatter', ('0', 'sum', torch.bfloat16)) len:4:CollBucket(collectives=[reduce_scatter_tensor, reduce_scatter_tensor_1, reduce_scatter_tensor_2, reduce_scatter_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=6291456)
coll:reduce_scatter_tensor hiding_nodes:OrderedSet([mm, addmm_default])
coll:reduce_scatter_tensor_1 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_2 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_3 hiding_nodes:OrderedSet([])
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @Lucaskabela

IvanKobzarev committed:
[inductor] bucketing stats logging

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[inductor] bucketing stats logging"


More logging for bucketing to compare runs 

Example of logged tlparse:
```
bucket key stats ('all_gather', ('0',)): 4 in 1 buckets of total:4
bucket key stats ('reduce_scatter', ('0', 'sum', torch.bfloat16)): 4 in 1 buckets of total:4

bucket[0] key:('all_gather', ('0',)) len:4:CollBucket(collectives=[all_gather_into_tensor, all_gather_into_tensor_1, all_gather_into_tensor_2, all_gather_into_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=12582912)
coll:all_gather_into_tensor hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_1 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_2 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_3 hiding_nodes:OrderedSet([])
bucket[1] key:('reduce_scatter', ('0', 'sum', torch.bfloat16)) len:4:CollBucket(collectives=[reduce_scatter_tensor, reduce_scatter_tensor_1, reduce_scatter_tensor_2, reduce_scatter_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=6291456)
coll:reduce_scatter_tensor hiding_nodes:OrderedSet([mm, addmm_default])
coll:reduce_scatter_tensor_1 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_2 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_3 hiding_nodes:OrderedSet([])
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[inductor] bucketing stats logging"


More logging for bucketing to compare runs 

Example of logged tlparse:
```
bucket key stats ('all_gather', ('0',)): 4 in 1 buckets of total:4
bucket key stats ('reduce_scatter', ('0', 'sum', torch.bfloat16)): 4 in 1 buckets of total:4

bucket[0] key:('all_gather', ('0',)) len:4:CollBucket(collectives=[all_gather_into_tensor, all_gather_into_tensor_1, all_gather_into_tensor_2, all_gather_into_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=12582912)
coll:all_gather_into_tensor hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_1 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_2 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_3 hiding_nodes:OrderedSet([])
bucket[1] key:('reduce_scatter', ('0', 'sum', torch.bfloat16)) len:4:CollBucket(collectives=[reduce_scatter_tensor, reduce_scatter_tensor_1, reduce_scatter_tensor_2, reduce_scatter_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=6291456)
coll:reduce_scatter_tensor hiding_nodes:OrderedSet([mm, addmm_default])
coll:reduce_scatter_tensor_1 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_2 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_3 hiding_nodes:OrderedSet([])
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[inductor] bucketing stats logging"


More logging for bucketing to compare runs 

Example of logged tlparse:
```
bucket key stats ('all_gather', ('0',)): 4 in 1 buckets of total:4
bucket key stats ('reduce_scatter', ('0', 'sum', torch.bfloat16)): 4 in 1 buckets of total:4

bucket[0] key:('all_gather', ('0',)) len:4:CollBucket(collectives=[all_gather_into_tensor, all_gather_into_tensor_1, all_gather_into_tensor_2, all_gather_into_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=12582912)
coll:all_gather_into_tensor hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_1 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_2 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_3 hiding_nodes:OrderedSet([])
bucket[1] key:('reduce_scatter', ('0', 'sum', torch.bfloat16)) len:4:CollBucket(collectives=[reduce_scatter_tensor, reduce_scatter_tensor_1, reduce_scatter_tensor_2, reduce_scatter_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=6291456)
coll:reduce_scatter_tensor hiding_nodes:OrderedSet([mm, addmm_default])
coll:reduce_scatter_tensor_1 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_2 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_3 hiding_nodes:OrderedSet([])
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[inductor] bucketing stats logging"


More logging for bucketing to compare runs 

Example of logged tlparse:
```
bucket key stats ('all_gather', ('0',)): 4 in 1 buckets of total:4
bucket key stats ('reduce_scatter', ('0', 'sum', torch.bfloat16)): 4 in 1 buckets of total:4

bucket[0] key:('all_gather', ('0',)) len:4:CollBucket(collectives=[all_gather_into_tensor, all_gather_into_tensor_1, all_gather_into_tensor_2, all_gather_into_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=12582912)
coll:all_gather_into_tensor hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_1 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_2 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_3 hiding_nodes:OrderedSet([])
bucket[1] key:('reduce_scatter', ('0', 'sum', torch.bfloat16)) len:4:CollBucket(collectives=[reduce_scatter_tensor, reduce_scatter_tensor_1, reduce_scatter_tensor_2, reduce_scatter_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=6291456)
coll:reduce_scatter_tensor hiding_nodes:OrderedSet([mm, addmm_default])
coll:reduce_scatter_tensor_1 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_2 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_3 hiding_nodes:OrderedSet([])
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[inductor] bucketing stats logging"


More logging for bucketing to compare runs 

Example of logged tlparse:
```
bucket key stats ('all_gather', ('0',)): 4 in 1 buckets of total:4
bucket key stats ('reduce_scatter', ('0', 'sum', torch.bfloat16)): 4 in 1 buckets of total:4

bucket[0] key:('all_gather', ('0',)) len:4:CollBucket(collectives=[all_gather_into_tensor, all_gather_into_tensor_1, all_gather_into_tensor_2, all_gather_into_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=12582912)
coll:all_gather_into_tensor hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_1 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_2 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_3 hiding_nodes:OrderedSet([])
bucket[1] key:('reduce_scatter', ('0', 'sum', torch.bfloat16)) len:4:CollBucket(collectives=[reduce_scatter_tensor, reduce_scatter_tensor_1, reduce_scatter_tensor_2, reduce_scatter_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=6291456)
coll:reduce_scatter_tensor hiding_nodes:OrderedSet([mm, addmm_default])
coll:reduce_scatter_tensor_1 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_2 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_3 hiding_nodes:OrderedSet([])
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

eellison commented on torch/_inductor/fx_passes/overlap_preserving_bucketer.py:
```diff
@@ -272,7 +276,18 @@ def _add_hiding_interval_constraints(self) -> None:
+        for start in self.collective_info:
+            pg = get_group_name(start)
+            if is_all_gather(start) and pg not in checked_pgs:
+                if is_fsdp_all_gather(start):
```
this function is not defined anywhere ? is this being used/tested ?

eellison reviewed (commented):
looks good other than unused identify_fsdp_group_names -maybe from bad split of pr

IvanKobzarev committed:
Update on "[inductor] bucketing stats logging"


More logging for bucketing to compare runs 

Example of logged tlparse:
```
bucket key stats ('all_gather', ('0',)): 4 in 1 buckets of total:4
bucket key stats ('reduce_scatter', ('0', 'sum', torch.bfloat16)): 4 in 1 buckets of total:4

bucket[0] key:('all_gather', ('0',)) len:4:CollBucket(collectives=[all_gather_into_tensor, all_gather_into_tensor_1, all_gather_into_tensor_2, all_gather_into_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=12582912)
coll:all_gather_into_tensor hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_1 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_2 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_3 hiding_nodes:OrderedSet([])
bucket[1] key:('reduce_scatter', ('0', 'sum', torch.bfloat16)) len:4:CollBucket(collectives=[reduce_scatter_tensor, reduce_scatter_tensor_1, reduce_scatter_tensor_2, reduce_scatter_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=6291456)
coll:reduce_scatter_tensor hiding_nodes:OrderedSet([mm, addmm_default])
coll:reduce_scatter_tensor_1 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_2 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_3 hiding_nodes:OrderedSet([])
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[inductor] bucketing stats logging"


More logging for bucketing to compare runs 

Example of logged tlparse:
```
bucket key stats ('all_gather', ('0',)): 4 in 1 buckets of total:4
bucket key stats ('reduce_scatter', ('0', 'sum', torch.bfloat16)): 4 in 1 buckets of total:4

bucket[0] key:('all_gather', ('0',)) len:4:CollBucket(collectives=[all_gather_into_tensor, all_gather_into_tensor_1, all_gather_into_tensor_2, all_gather_into_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=12582912)
coll:all_gather_into_tensor hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_1 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_2 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_3 hiding_nodes:OrderedSet([])
bucket[1] key:('reduce_scatter', ('0', 'sum', torch.bfloat16)) len:4:CollBucket(collectives=[reduce_scatter_tensor, reduce_scatter_tensor_1, reduce_scatter_tensor_2, reduce_scatter_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=6291456)
coll:reduce_scatter_tensor hiding_nodes:OrderedSet([mm, addmm_default])
coll:reduce_scatter_tensor_1 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_2 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_3 hiding_nodes:OrderedSet([])
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[inductor] bucketing stats logging"


More logging for bucketing to compare runs 

Example of logged tlparse:
```
bucket key stats ('all_gather', ('0',)): 4 in 1 buckets of total:4
bucket key stats ('reduce_scatter', ('0', 'sum', torch.bfloat16)): 4 in 1 buckets of total:4

bucket[0] key:('all_gather', ('0',)) len:4:CollBucket(collectives=[all_gather_into_tensor, all_gather_into_tensor_1, all_gather_into_tensor_2, all_gather_into_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=12582912)
coll:all_gather_into_tensor hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_1 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_2 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_3 hiding_nodes:OrderedSet([])
bucket[1] key:('reduce_scatter', ('0', 'sum', torch.bfloat16)) len:4:CollBucket(collectives=[reduce_scatter_tensor, reduce_scatter_tensor_1, reduce_scatter_tensor_2, reduce_scatter_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=6291456)
coll:reduce_scatter_tensor hiding_nodes:OrderedSet([mm, addmm_default])
coll:reduce_scatter_tensor_1 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_2 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_3 hiding_nodes:OrderedSet([])
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[inductor] bucketing stats logging"


More logging for bucketing to compare runs 

Example of logged tlparse:
```
bucket key stats ('all_gather', ('0',)): 4 in 1 buckets of total:4
bucket key stats ('reduce_scatter', ('0', 'sum', torch.bfloat16)): 4 in 1 buckets of total:4

bucket[0] key:('all_gather', ('0',)) len:4:CollBucket(collectives=[all_gather_into_tensor, all_gather_into_tensor_1, all_gather_into_tensor_2, all_gather_into_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=12582912)
coll:all_gather_into_tensor hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_1 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_2 hiding_nodes:OrderedSet([])
coll:all_gather_into_tensor_3 hiding_nodes:OrderedSet([])
bucket[1] key:('reduce_scatter', ('0', 'sum', torch.bfloat16)) len:4:CollBucket(collectives=[reduce_scatter_tensor, reduce_scatter_tensor_1, reduce_scatter_tensor_2, reduce_scatter_tensor_3], bucketed_start=None, bucketed_wait=None, total_bytes=6291456)
coll:reduce_scatter_tensor hiding_nodes:OrderedSet([mm, addmm_default])
coll:reduce_scatter_tensor_1 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_2 hiding_nodes:OrderedSet([])
coll:reduce_scatter_tensor_3 hiding_nodes:OrderedSet([])
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

IvanKobzarev commented:
@pytorchbot merge
